### PR TITLE
feat(context): add standalone examples to illustrate features provided by @loopback/context

### PR DESCRIPTION
**Pull request by raymondfeng**
This PR adds a set of standalone examples to `packages/context` to illustrate common recipes for various features offered by our IoC container and dependency injection framework.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

**Comment by raymondfeng** [comment]
I'm not sure which of the following options is better to host these examples:

1. in `packages/context/src/__examples`
2. In `examples/context-features`

**Comment by bajtos** [comment]
> I'm not sure which of the following options is better to host these examples:
> 
> 1. in `packages/context/src/__examples__`
> 2. In `examples/context-features`

Both options are fine with me.

The downside of using `packages/context/src/__examples__`: example file are importing from `..` instead of `@loopback/context`, therefore LB users cannot simply copy an example file from GitHub to their local project as-is to further tweak it & play with it - they have to fix the import statement first.

**Comment by raymondfeng** [comment]
See https://github.com/strongloop/loopback-next/blob/add-context-examples/packages/context/src/__tests__/acceptance/examples.acceptance.ts

**Comment by raymondfeng** [comment]
Being able to copy and paste is great. I moved the code into `examples/context` as a separate package.

File: examples/context/src/__tests__/acceptance/examples.acceptance.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+describe('context examples', () => {
+  let logs: string[] = [];
+  let originalLog = console.log;
+
+  before(disableConsoleLog);
+
+  it('runs all examples', async () => {
+    await main();
+    expect(replaceDates(logs)).to.eql(replaceDates(expectedLogs));
```
[the remark this answers is not in the document]
Comment (raymondfeng) [reply examples/context/src/__tests__/acceptance/examples.acceptance.ts]: I improved the test to capture `console.error` as well.

File: examples/context/LICENSE [hunk 2]
```diff
@@ -0,0 +1,25 @@
+Copyright (c) IBM Corp. 2018,2019. All Rights Reserved.
```
Comment (nabdelgadir) [inline examples/context/LICENSE]: `2018,2019` -> `2019`

File: examples/context/package.json [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  "dependencies": {
+    "@loopback/context": "^1.16.0"
+  },
+  "devDependencies": {
+    "@loopback/build": "^1.5.5",
+    "@loopback/testlab": "^1.2.10",
+    "@loopback/tslint-config": "^2.0.4",
+    "@types/node": "^10.11.2",
+    "tslint": "^5.16.0",
+    "typescript": "^3.4.5"
```
Comment (nabdelgadir) [inline examples/context/package.json]: `"typescript": "~3.5.1"`?

File: examples/context/src/context-chain.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  console.log(greeter.greet('John'));
+
+  // Set `prefix` at request context level
+  requestCtx.bind('prefix').to(requestCtx.name);
+  greeter = await requestCtx.get<Greeter>(greeterBinding.key);
+  // Now the request context prefix is used
+  console.log(greeter.greet('John'));
+
+  // Get a greeter from app context
+  greeter = await requestCtx.get<Greeter>(greeterBinding.key);
```
Comment (nabdelgadir) [inline examples/context/src/context-chain.ts]: Should this one be `greeter = await appCtx.get<Greeter>(greeterBinding.key);`?

Comment (raymondfeng) [reply examples/context/src/context-chain.ts]: Good catch!

**Pull request merged**